pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -582,9 +586,18 @@ func (cb *ClusterBuilder) setUseDownstreamProtocol(mc *clusterWrapper) {
 		mc.httpProtocolOptions = &http.HttpProtocolOptions{}
 	}
 	options := mc.httpProtocolOptions
+	// Inherit the HTTP1.x protocol options from the explicit HTTP1.x config.
+	explicitHTTP1Options := &core.Http1ProtocolOptions{}
+	if explicitConfig, ok := options.GetUpstreamProtocolOptions().(*http.HttpProtocolOptions_ExplicitHttpConfig_); ok {
+		protocolConfig, ok := explicitConfig.ExplicitHttpConfig.ProtocolConfig.(*http.HttpProtocolOptions_ExplicitHttpConfig_HttpProtocolOptions)
+		if ok && protocolConfig.HttpProtocolOptions != nil {
+			// Use existing HTTP/1.x options if available
+			explicitHTTP1Options = protocolConfig.HttpProtocolOptions
+		}
+	}
```

Comment:
```suggestion
	if opts := options.GetExplicitHttpConfig().GetHttpProtocolOptions(); opts != nil {
		explicitHTTP1Options = opts
	}
```

proto provides some helpers to make this more ergonomic. Same behavior though